### PR TITLE
enhance OpenFold 1.0.1 for standalone usage

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

- replace `OpenMM` dep with its `CUDA` variant
- add cmd line tools like `run_pretrained_openfold.py`
- add openfold and alphafold params
- add `scripts` directory
- add extension `flash_attn` for better performance

### CODE REVIEW

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node403.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7282 16-Core Processor, 1 x NVIDIA NVIDIA A100-PCIE-40GB, 515.48.07, Python 3.6.8
See https://gist.github.com/c58dcac88d4b98313fdef787892777cc for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node401.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7282 16-Core Processor, 1 x NVIDIA NVIDIA A100-PCIE-40GB, 515.48.07, Python 3.6.8
See https://gist.github.com/96789d13fb37aa818490f510f3e77154 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node250.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, 1 x NVIDIA Tesla P100-PCIE-16GB, 515.48.07, Python 3.6.8
See https://gist.github.com/b2cb36120e24aa4406441e098852ad78 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node403.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7282 16-Core Processor, 1 x NVIDIA NVIDIA A100-PCIE-40GB, 515.48.07, Python 3.6.8
See https://gist.github.com/1d33f0f7113504ce2efa430d02967f1b for a full test report.

**Comment by smoors** [comment]
the failing OpenCL tests for OpenMM were due to NVIDIA MPS service being run.
without MPS, all tests passed.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb [hunk 1]
```diff
@@ -80,13 +86,34 @@ exts_list = [
...
+    # 'ln -s %s/stereo_chemical_props.txt %s' % (_resources_target_dir, _resources_dir),
+    # 'ln -s %s/openfold_params %s' % (_resources_target_dir, _resources_dir),
+    # 'ln -s %s/params %s' % (_resources_target_dir, _resources_dir),
+]
+
+fix_python_shebang_for = ['bin/*.py', 'scripts/*.py']
+
+modextrapaths = {
+    'PATH': 'scripts',
+    'PYTHONPATH': '',
```
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb:106]: indeed, `run_pretrained_openfold.py` imports modules in the scripts directory, e.g.
```
from scripts.utils import add_data_args
```

File: easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb [hunk 2]
```diff
@@ -33,12 +36,15 @@ dependencies = [
...
     # https://github.com/aqlaboratory/openfold/blob/v1.0.0/scripts/install_third_party_dependencies.sh#L20-L24
     # https://github.com/aqlaboratory/openfold/blob/v1.0.0/lib/openmm.patch
-    ('OpenMM', '7.5.1', '-DeepMind-patch'),
+    ('OpenMM', '7.5.1', '-CUDA-%(cudaver)s-DeepMind-patch'),
 ]
 
 use_pip = True
 
 exts_list = [
+    ('flash_attn', '0.2.8', {
```
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb:45]: it's an optional dependency that should greatly speed up some parts of the calculation (see FlashAttention in the readme)

File: easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb (hunk 1 above)
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb:106]: we could move the scripts directory into the bin directory, that may be safer?

[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb:106]: I'll move the scripts directory inside bin

File: easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb (hunk 2 above)
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.1-foss-2021a-CUDA-11.3.1.eb:42]: I'll add a comment

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (2 easyconfigs in total)
node3300.joltik.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), 1 x NVIDIA Tesla V100-SXM2-32GB, 525.60.13, Python 3.6.8
See https://gist.github.com/77eb148d490614e8b2d9ad9d4660d480 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
node3906.accelgor.os - Linux RHEL 8.6, x86_64, AMD EPYC 7413 24-Core Processor (zen3), 1 x NVIDIA [Unknown Error], 525.60.13, Python 3.6.8
See https://gist.github.com/6494cdb2f4575fbed47f157f36b8c9b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3307.joltik.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), 1 x NVIDIA Tesla V100-SXM2-32GB, 525.60.13, Python 3.6.8
See https://gist.github.com/1efeac918a1f21c4e224f61f027db355 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3903.accelgor.os - Linux RHEL 8.6, x86_64, AMD EPYC 7413 24-Core Processor (zen3), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 525.60.13, Python 3.6.8
See https://gist.github.com/646bb0b17baa97e4800d20087f497941 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Pull request merged**